### PR TITLE
Adds an interaction between two SMs when they collide.

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
When a supermatter shard collides with another it will deal 20% integrity damage and massively increase the EER.

## Why It's Good For The Game
A supermatter dusting another supermatter should have a massive reaction, all the energy contained with one crystal is instantly being released. 
This allows for interesting sabotages that involve throwing shards into the crystal to hyper energize it, giving engineers a challenge to fix compared to current sabotage which can be fixed with N2 and a fire extinguisher. Currently engine sabotage is barely worth a hijacker/nukies time because of engineers being able to fix it relatively easily and the minimal damage it courses. 
This also makes making more SMs slightly more dangerous because currently if a shard is delamming you can just throw it into the main crystal and it is gone.

## Images of changes
ezgif-7-8c56b242ec98

## Changelog
:cl:
add: throwing an SM shard into another will cause one of the shards to have a massive boost in energy and a drop in integrity. 
/:cl:

### CODE REVIEW

**Comment by Adrer** [comment]
Now I am become death

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -752,6 +752,12 @@
 				suspicion = "last touched by [AM.fingerprintslast]"
 			message_admins("[src] has consumed [AM], [suspicion] [ADMIN_JMP(src)].")
 			investigate_log("has consumed [AM] - [suspicion].", "supermatter")
+			if(istype(AM, /obj/machinery/power/supermatter_crystal))
+				power += 5000//releases A LOT of power
+				matter_power += 500000
+				damage += 180//drops the integrety by 20%
+				AM.visible_message("<span class='danger'>[AM] smacks into [src], rapidly flashes into pure energy. The energy inside [src] begins superradiance scattering!</span>", null,\
```
Comment (S34NW) [inline code/modules/power/supermatter/supermatter.dm:759]:
```suggestion
				AM.visible_message("<span class='danger'>[AM] smacks into [src], rapidly flashing blasts of pure energy. The energy inside [src] undergoes superradiance scattering!</span>", null,\
```

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -752,6 +752,12 @@
...
+				"<span class='italics'>You hear a loud crack as you are washed with a wave of heat.</span>")
```
Comment (S34NW) [inline code/modules/power/supermatter/supermatter.dm:760]:
```suggestion
				"<span class='italics'>You hear a loud crack as a wave of heat washes over you.</span>")
```

**Review by S34NW — COMMENTED** [review]
few things

**Review by dearmochi — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]
Code seems fine by me

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -752,6 +752,12 @@
 				suspicion = "last touched by [AM.fingerprintslast]"
 			message_admins("[src] has consumed [AM], [suspicion] [ADMIN_JMP(src)].")
 			investigate_log("has consumed [AM] - [suspicion].", "supermatter")
+			if(istype(AM, /obj/machinery/power/supermatter_crystal))
+				power += 5000//releases A LOT of power
+				matter_power += 500000
+				damage += 180//drops the integrety by 20%
+				AM.visible_message("<span class='danger'>[AM] smacks into [src], rapidly flashing blasts of pure energy. The energy inside [src] undergoes superradiance scattering!</span>", null,\
+				"<span class='italics'>You hear a loud crack as a wave of heat washes over you.</span>")
```
Comment (SabreML) [inline code/modules/power/supermatter/supermatter.dm:760]: Very nitpicky, but it's after a comma so no `\` needed.
```suggestion
				AM.visible_message("<span class='danger'>[AM] smacks into [src], rapidly flashing blasts of pure energy. The energy inside [src] undergoes superradiance scattering!</span>", null,
				"<span class='italics'>You hear a loud crack as a wave of heat washes over you.</span>")
```

**Pull request merged**